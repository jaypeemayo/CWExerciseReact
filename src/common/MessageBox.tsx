import Modal from 'react-modal';
import * as React from "react";

export interface IMessageBox {
   title: string,
   buttonSet: any,
}

const customStyles = {
    content: {
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      marginRight: '-50%',
      transform: 'translate(-50%, -50%)',
    },
  };

  
export const MessageBox = (props: React.PropsWithChildren<IMessageBox>) => {
    return <Modal style={customStyles}  isOpen={true}>
        <div role="dialog"  id="messsageBox" aria-labelledby="messageBoxLabel"  aria-modal="true">
            <h1 id="messageBoxLabel">
                {props.title}
            </h1>
            <p>
                {props.children}
            </p>
            <div>{props.buttonSet}</div>

        </div>
    </Modal>

}
