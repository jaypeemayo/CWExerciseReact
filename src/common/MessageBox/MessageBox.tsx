import Modal from 'react-modal';
import * as React from "react";

export interface IMessageBox {
    title: string,
    buttonSet: any,
}

const customStyles = {
    content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
        padding:'0px',
        width:'400px'
    },
};

export const MessageBox = (props: React.PropsWithChildren<IMessageBox>) => {
    return <Modal style={customStyles} isOpen={true} ariaHideApp={false} >
        <div className="modal-content" role="dialog" id="messsageBox" aria-labelledby="messageBoxLabel">
            <div className="modal-header">
                <h5 id="messageBoxLabel">
                    {props.title}
                </h5>
            </div>
            <div className="modal-body">
                <p>
                    {props.children}
                </p>
            </div>
            <div className="modal-footer">{props.buttonSet}</div>

        </div>
    </Modal>

}
