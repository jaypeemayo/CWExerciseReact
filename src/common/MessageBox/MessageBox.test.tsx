import { render } from "@testing-library/react";
import * as React from "react";
import { MessageBox } from "./MessageBox";

jest.mock('react-modal', () => {
    const React = require('react');
    const TestReactModal = require('../../../tests/__mocks__/mockReactModal');
    return TestReactModal.default;
  });
  
  
test("Test snapshot", () => {
    const {container, unmount} = render(
        <MessageBox
            title={"Test Title"}
            buttonSet={<button>test Buttonset</button>}
        />,
    );

    expect(container).toMatchSnapshot();
    unmount();


});


