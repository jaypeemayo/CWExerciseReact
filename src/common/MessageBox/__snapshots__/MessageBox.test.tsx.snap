// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot 1`] = `
<div>
  <div>
    <div
      aria-labelledby="messageBoxLabel"
      class="modal-content"
      id="messsageBox"
      role="dialog"
    >
      <div
        class="modal-header"
      >
        <h5
          id="messageBoxLabel"
        >
          Test Title
        </h5>
      </div>
      <div
        class="modal-body"
      >
        <p />
      </div>
      <div
        class="modal-footer"
      >
        <button>
          test Buttonset
        </button>
      </div>
    </div>
  </div>
</div>
`;
